deploy/chart/templates/0000_50_18-service-monitor.yaml

```diff
@@ -0,0 +1,102 @@
+- apiVersion: monitoring.coreos.com/v1
+  kind: ServiceMonitor
+  metadata:
+    name: olm-operator
+    namespace: {{ .Values.namespace }}
+    labels:
+      app: olm-operator
+  spec:
+    jobLabel: k8s-app
+    endpoints:
+    - port: http-metrics
+      interval: 30s
+      scheme: http
+      bearerTokenFile: /var/run/secrets/kubernetes.io/serviceaccount/token
+    selector:
+      matchLabels:
+        app: olm-operator
+    namespaceSelector:
+      matchNames: {{ .Values.namespace }}
+---
+- apiVersion: v1
+  kind: Service
+  metadata:
+    name: olm-operator
+    namespace: {{ .Values.namespace }}
+    labels:
+      app: olm-operator
+  spec:
+    ports:
+    - name: http-metrics
+      port: 8081
+      protocol: TCP
+      targetPort: http
+    selector:
+      app: olm-operator
+---
+- apiVersion: monitoring.coreos.com/v1
+  kind: ServiceMonitor
+  metadata:
+    name: catalog-operator
+    namespace: {{ .Values.namespace }}
+    labels:
+      app: catalog-operator
+  spec:
+    jobLabel: k8s-app
+    endpoints:
+    - port: http-metrics2
+      interval: 30s
+      scheme: http
+      bearerTokenFile: /var/run/secrets/kubernetes.io/serviceaccount/token
+    selector:
+      matchLabels:
+        app: catalog-operator
+    namespaceSelector:
+      matchNames: {{ .Values.namespace }}
+---
+- apiVersion: v1
+  kind: Service
+  metadata:
+    name: catalog-operator
+    namespace: {{ .Values.namespace }}
+    labels:
+      app: catalog-operator
+  spec:
+    ports:
+    - name: http-metrics2
```

Comment:
we don't need to disambiguate since the endpoints are defined in different services. you don't need the name field at all